src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -383,6 +389,43 @@ private void AddAnonymousTypeName(INamedTypeSymbol symbol)
             }
         }
 
+        /// <summary>
+        /// Returns true if tuple type syntax can be used to refer to the tuple type without loss of information.
+        /// For example, it cannot be used when extension tuple is using non-default friendly names. 
+        /// </summary>
+        /// <param name="symbol"></param>
+        /// <returns></returns>
+        private bool CanUseTupleTypeName(INamedTypeSymbol symbol)
+        {
+            INamedTypeSymbol currentUnderlying = symbol.TupleUnderlyingType;
+
+            if (currentUnderlying.Arity == TupleTypeSymbol.RestPosition)
+            {
+                do
+                {
+                    var currentTuple = (INamedTypeSymbol)currentUnderlying.TypeArguments[TupleTypeSymbol.RestPosition - 1];
+                    Debug.Assert(currentTuple.IsTupleType);
+
+                    var elementNames = currentTuple.TupleElementNames;
+                    if (!elementNames.IsDefault)
+                    {
+                        for (int i = 0; i < elementNames.Length; i++)
+                        {
+                            if (elementNames[i] != TupleTypeSymbol.TupleMemberName(i + 1))
+                            {
+                                return false;
+                            }
+                        }
+                    }
+
+                    currentUnderlying = currentTuple.TupleUnderlyingType;
+                }
+                while (currentUnderlying.Arity == TupleTypeSymbol.RestPosition);
+            }
+
+            return true;
+        }
+
```

Comment:
That issue is different. In there, both types are tuples and changing their display won't change anything, the display will still look the same. That diagnostics should include containing assembly for both types to make the error clearer.

---
In reply to: [62075672](https://github.com/dotnet/roslyn/pull/11047#discussion_r62075672) [](ancestors = 62075672)